core/src/main/java/nl/nn/adapterframework/jms/JmsListenerBase.java

```diff
@@ -132,51 +132,41 @@ public void close() {
 	 * externally.
 	 *
 	 * @param rawMessage - Original message received, can not be <code>null</code>
-	 * @param threadContext - Thread context to be populated, can not be <code>null</code>
+	 * @return
 	 */
-	public String getIdFromRawMessage(javax.jms.Message rawMessage, Map<String, Object> threadContext) throws ListenerException {
-		TextMessage message = null;
-		try {
-			message = (TextMessage) rawMessage;
-		} catch (ClassCastException e) {
-			log.error("message received by listener on ["+ getDestinationName()+ "] was not of type TextMessage, but ["+rawMessage.getClass().getName()+"]", e);
-			return null;
-		}
-		return retrieveIdFromMessage(message, threadContext);
-	}
-
-	protected String retrieveIdFromMessage(javax.jms.Message message, Map<String, Object> threadContext) throws ListenerException {
+	public Map<String, Object> populateContextFromMessage(javax.jms.Message rawMessage) throws ListenerException {
+		Map<String, Object> messageContext = new HashMap<>();
```

Comment:
Dit is niet de MessageContext, maar meer een soort sessie die buiten de pipeline leeft.

Reply:
Of een threadcontext die buiten de thread leeft.

Ik kan het hernoemen naar `messageProperties`?